src/runtime_src/core/common/query_requests.h

```diff
@@ -535,6 +541,54 @@ struct xclbin_uuid : request
   get(const device*) const = 0;
 };
 
+struct group_topology : request
+{
+  using result_type = std::vector<char>;
+  static const key_type key = key_type::group_topology;
+
+  virtual boost::any
+  get(const device*) const = 0;
+};
+
+struct temp_by_mem_topology : request
+{
+  using result_type = std::vector<char>;
+  static const key_type key = key_type::temp_by_mem_topology;
+
+  virtual boost::any
+  get(const device*) const = 0;
+};
+
+struct memstat : request
+{
+  using result_type = std::vector<char>;
```

Comment:
This looks like raw (char*) data is returned.  You sure you didn't flip memstat and memstat_raw?